google/resource_cloudfunctions_function.go

```diff
@@ -0,0 +1,462 @@
+		switch getOpt.EventTrigger.EventType {
+		//From https://github.com/google/google-api-go-client/blob/master/cloudfunctions/v1/cloudfunctions-gen.go#L335
+		case "providers/cloud.pubsub/eventTypes/topic.publish":
+			d.Set("trigger_topic", extractLastResourceFromUri(getOpt.EventTrigger.Resource))
+		case "providers/cloud.storage/eventTypes/object.change":
+			d.Set("trigger_bucket", extractLastResourceFromUri(getOpt.EventTrigger.Resource))
+		}
+	}
+	d.Set("project", funcProject)
+
+	return nil
+}
+
+func resourceCloudFunctionsUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	service := config.clientCloudFunctions
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	d.Partial(true)
+
+	function := cloudfunctions.CloudFunction{
+		Name: createCloudFunctionsPathString(CLOUDFUNCTIONS_FULL_NAME, project, region, d.Get("name").(string)),
+	}
+
+	_, err = service.Projects.Locations.Functions.Get(function.Name).Do()
+
+	if err != nil {
+		return fmt.Errorf("Function %s doesn't exists.", d.Get("name").(string))
+	}
+
+	if d.HasChange("labels") {
+		function.Labels = expandLabels(d)
+
+		op, err := service.Projects.Locations.Functions.Patch(function.Name, &function).
+			UpdateMask("labels").Do()
+
+		if err != nil {
+			return fmt.Errorf("Error when updating labels: %s", err)
+		}
+
+		d.SetPartial("labels")
+		err = cloudFunctionsOperationWait(config.clientCloudFunctions, op, "Updating CloudFunctions Function Labels")
+		if err != nil {
+			return err
+		}
+	}
+	configUpdate := false
+	var updateMaskArr []string
+	var partialArr []string
+	if d.HasChange("memory") {
+		memory := d.Get("memory").(int)
+
+		if FUNCTION_ALLOWED_MEMORY[memory] != true {
+			return fmt.Errorf("Allowed values for memory are: 128MB, 256MB, 512MB, 1024MB, and 2048MB. "+
+				"Got %d", memory)
+		}
+		function.AvailableMemoryMb = int64(memory)
+		updateMaskArr = append(updateMaskArr, "availableMemoryMb")
+		partialArr = append(partialArr, "memory")
+		configUpdate = true
+	}
+
+	if d.HasChange("description") {
+		function.Description = d.Get("description").(string)
+		updateMaskArr = append(updateMaskArr, "description")
+		partialArr = append(partialArr, "description")
+		configUpdate = true
+	}
+
+	if d.HasChange("timeout") {
+		function.Timeout = fmt.Sprintf("%vs", d.Get("timeout").(int))
+		updateMaskArr = append(updateMaskArr, "timeout")
+		partialArr = append(partialArr, "timeout")
+		configUpdate = true
+	}
+
+	if configUpdate {
+		log.Printf("[DEBUG] Send Patch CloudFunction Configuration request: %#v", function)
+		updateMask := strings.Join(updateMaskArr, ",")
+		op, err := service.Projects.Locations.Functions.Patch(function.Name, &function).
+			UpdateMask(updateMask).Do()
+
+		if err != nil {
+			return fmt.Errorf("Error while updating cloudfunction configuration: %s", err)
+		}
+		for i := range partialArr {
```

Comment:
This would only be necessary if you were sending multiple requests. Since you're only sending one, no need for `SetPartial`

Reply:
Fixed